### PR TITLE
Fix subject encoding error with Python3.4 #88

### PR DESCRIPTION
**Pull request by harobed**
Fix subject encoding error with Python3.4 #88

### CODE REVIEW

**Comment by decaz** [comment]
The bug seems to be fixed by this PR. Why not to merge it?

**Comment by iorlas** [comment]
Because it looks like it almost dead in activity

**Comment by wlach** [comment]
(looking at this as part of the PyCon US 2024 sprints)

@stephane-klein @harobed We recently moved this to pallets-eco and I've been looking at merging old fixes / issues. As far as I can tell this looks approximately correct, however, your work needs to be reapplied against main. Are you interested in updating this PR or creating a new one? If so please do so and tag me.

File: tests.py [hunk 1]
```diff
@@ -428,7 +428,12 @@ def test_unicode_subject(self):
         msg = Message(subject=make_lazy_string(lambda a: a, u"sübject"),
                       sender='from@example.com',
                       recipients=["to@example.com"])
-        self.assertIn('=?utf-8?q?s=C3=BCbject?=', msg.as_string())
+        self.assertIn(b'=?utf-8?q?s=C3=BCbject?=', msg.as_bytes())
```
Comment (wlach) [inline tests.py:431]: Shouldn't we still test the `as_string()` case? I could be missing something.

**Comment by davidism** [comment]
If `email.Message.as_bytes()` is failing on something it shouldn't, that seems to be an issue to report to Python's smtplib. 3.8 is the lowest supported version in 2024, and it does not appear to have an issue.

**Pull request closed without merging**